app/lib/payment_gateways/stripe_crypt.rb

```diff
@@ -15,35 +15,40 @@ def initialize(user)
     end
 
     def update!(params)
-      stripe_params = params.require(:stripe)
-      customer = create_stripe_customer(stripe_params)
-      update_account(customer, stripe_params)
-    rescue Stripe::CardError => error
-      report_error(error.message)
+      payment_method_id = params.require(:stripe).require(:payment_method_id)
+      payment_method = Stripe::PaymentMethod.retrieve(payment_method_id, api_key)
+      card = payment_method.card
+
+      account.credit_card_expires_on_month = card.exp_month
+      account.credit_card_expires_on_year = card.exp_year
+      account.credit_card_partial_number = card.last4
+      account.credit_card_auth_code = payment_method.customer
```

Comment:
Should we be creating a `PaymentDetail` here instead of using these (deprecated) helper methods?